src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceEvaluator.cs

```diff
@@ -0,0 +1,1117 @@
+            }
+        }
+
+        private static bool TryValidateDailyRecurrencePattern(TimeWindowFilterSettings settings, out string paramName, out string reason)
+        {
+            Debug.Assert(settings.Recurrence.Pattern.Interval >  0);
+
+            TimeSpan intervalDuration = TimeSpan.FromDays(settings.Recurrence.Pattern.Interval);
+
+            TimeSpan timeWindowDuration = settings.End.Value - settings.Start.Value;
+
+            //
+            // Time window duration must be shorter than how frequently it occurs
+            if (timeWindowDuration > intervalDuration)
+            {
+                paramName = $"{nameof(settings.End)}";
+
+                reason = TimeWindowDurationOutOfRange;
+
+                return false;
+            }
+
+            //
+            // No required parameter for "Daily" pattern
+            // "Start" is always a valid first occurrence for "Daily" pattern
+
+            paramName = null;
+
+            reason = null;
+
+            return true;
+        }
+
+        private static bool TryValidateWeeklyRecurrencePattern(TimeWindowFilterSettings settings, out string paramName, out string reason)
+        {
+            RecurrencePattern pattern = settings.Recurrence.Pattern;
+
+            Debug.Assert(pattern.Interval > 0);
+
+            TimeSpan intervalDuration = TimeSpan.FromDays(pattern.Interval * DayNumberOfWeek);
+
+            TimeSpan timeWindowDuration = settings.End.Value - settings.Start.Value;
+
+            //
+            // Time window duration must be shorter than how frequently it occurs
+            if (timeWindowDuration > intervalDuration || 
+                !IsDurationCompliantWithDaysOfWeek(timeWindowDuration, pattern.Interval, pattern.DaysOfWeek, pattern.FirstDayOfWeek))
+            {
+                paramName = $"{nameof(settings.End)}";
+
+                reason = TimeWindowDurationOutOfRange;
+
+                return false;
+            }
+
+            //
+            // Required parameters
+            if (!TryValidateDaysOfWeek(settings, out paramName, out reason))
+            {
+                return false;
+            }
+
+            //
+            // Check whether "Start" is a valid first occurrence
+            DateTimeOffset start = settings.Start.Value;
+
+            if (!pattern.DaysOfWeek.Any(day =>
+                day == start.DayOfWeek))
+            {
+                paramName = nameof(settings.Start);
+
+                reason = StartNotMatched;
+
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool TryValidateAbsoluteMonthlyRecurrencePattern(TimeWindowFilterSettings settings, out string paramName, out string reason)
+        {
+            RecurrencePattern pattern = settings.Recurrence.Pattern;
+
+            Debug.Assert(pattern.Interval > 0);
+
+            TimeSpan intervalDuration = TimeSpan.FromDays(pattern.Interval * MinDayNumberOfMonth);
```

Comment:
How does this work when someone sets the duration to 29 days for example, and their recurrence doesn't include february? Is that a valid scenario?

Reply:
If the recurrence pattern type is "AbsoluteMonthly" or "RelativeMonthly“ and the "Interval" is 1, the time window duration cannot be longer than 28 days. If the "Interval" is 2, then the time window duration cannot be longer than 48 days. This is the behavior of Outlook. You can check this in the Outlook app by creating a meeting and making it recurring. Please use the old Outlook where you can see the error message. If you use the new Outlook, it only tells you that it fails to create the event.
![image](https://github.com/microsoft/FeatureManagement-Dotnet/assets/141655842/7a6e57a8-e7e6-466e-8e08-5badbb45d61b)
![Uploading image.png…]()

Reply:
Thanks for the explanation! If that's the outlook behavior then I think it's reasonable to keep things consistent